parsec/core/fs/workspacefs/sync_transactions.py

```diff
@@ -45,6 +47,39 @@
     },
 }
 
+# Data class for change info
+
+
+@attr.s(frozen=True, auto_attribs=True)
+class ChangesAfterSync:
+    added_blocks: set[BlockAccess] | None = None
+    removed_blocks: set[BlockAccess] | None = None
+    added_entries: set[EntryID] | None = None
+    removed_entries: set[EntryID] | None = None
+
+    @classmethod
+    def from_manifests(
+        cls,
+        old_manifest: AnyRemoteManifest,
+        new_manifest: AnyRemoteManifest,
```

Comment:
Use a TypeVar

Reply:
Too painful as it would require to type more stuff more strictly in sync_transactions